agentseek/src/main/kotlin/io/github/agentseek/components/NoiseEmitterVisualComponent.kt

```diff
@@ -0,0 +1,28 @@
+package io.github.agentseek.components
+
+import io.github.agentseek.common.Circle2d
+import io.github.agentseek.common.TimerImpl
+import io.github.agentseek.core.GameObject
+import io.github.agentseek.view.animations.VFX
+import java.awt.Color
+import kotlin.time.Duration
+
+@Requires(NoiseEmitterComponent::class)
+class NoiseEmitterVisualComponent(gameObject: GameObject) : AbstractComponent(gameObject) {
+    private val timer = TimerImpl(500)
+    override fun init() {
+        timer.startTimer()
+    }
+
+    override fun onUpdate(deltaTime: Duration) {
+        gameObject.getComponent<NoiseEmitterComponent>()!!.getNoiseEmitterCollider()?.let {
```

Comment:
Here we're getting the component each frame, but it is not necessary. We can get it in the init just once.